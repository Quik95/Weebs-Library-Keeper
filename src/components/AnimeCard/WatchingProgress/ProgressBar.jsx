import React, { memo } from 'react';
import PropTypes from 'prop-types';

// MUI
import Tooltip from '@material-ui/core/Tooltip';

// styles
import clsx from 'clsx';
import useStyles from './WatchingProgress.styles';

const ProgressBar = memo(function ProgressBar({ className, tooltipText }) {
  const classes = useStyles();

  return (
    <Tooltip title={tooltipText}>
      <div className={clsx(classes.progress, classes[className])} />
    </Tooltip>
  );
});

ProgressBar.propTypes = {
  className: PropTypes.string.isRequired,
  tooltipText: PropTypes.string.isRequired,
};

export default ProgressBar;
