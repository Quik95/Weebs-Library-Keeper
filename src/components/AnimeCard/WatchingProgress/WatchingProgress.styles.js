import { makeStyles } from '@material-ui/core/styles';
import { amber } from '@material-ui/core/colors';

export default makeStyles(theme => ({
  progressBarContainer: {
    display: 'flex',
    flexWrap: 'nowrap',
    width: '100%',
    height: '5px',
    marginLeft: theme.spacing(2),
    position: 'relative',
  },
  progress: {
    height: '100%',
    position: 'absolute',
  },
  'progress--total': {
    backgroundColor: theme.palette.secondary.light,
    width: '100%',
    zIndex: 1,
  },
  'progress--watched': {
    backgroundColor: theme.palette.secondary.main,
    zIndex: 3,
    width: '20%',
  },
  'progress--aired': {
    backgroundColor: amber[700],
    zIndex: 2,
    width: '60%',
  },
  watchingProgress: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
}));
