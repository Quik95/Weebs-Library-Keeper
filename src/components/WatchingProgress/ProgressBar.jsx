import React, { memo } from 'react';
import PropTypes from 'prop-types';

// styles
import clsx from 'clsx';
import useStyles from './WatchingProgress.styles';

const ProgressBar = memo(function ProgressBar(props) {
  const classes = useStyles();

  return (
    <div className={classes.progressBarContainer}>
      <div className={clsx(classes['progress--watched'], classes.progress)} />
      <div className={clsx(classes['progress--aired'], classes.progress)} />
      <div className={clsx(classes['progress--total'], classes.progress)} />
    </div>
  );
});

ProgressBar.propTypes = {};

export default ProgressBar;
