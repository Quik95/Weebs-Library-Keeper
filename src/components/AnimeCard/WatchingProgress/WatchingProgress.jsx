import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';

// Custom components
import ProgressBar from './ProgressBar/ProgressBar';

// styles
import useStyles from './WatchingProgress.styles';

const WatchingProgress = memo(function WatchingProgress(props) {
  const classes = useStyles();

  return (
    <div className={classes.watchingProgress} data-testid="WatchingProgress">
      <Typography variant="caption">10/15</Typography>
      <div className={classes.progressBarContainer}>
        <ProgressBar
          className="progress--watched"
          tooltipText="Number of episodes you watched"
        />
        <ProgressBar
          className="progress--aired"
          tooltipText="Number of episodes aired"
        />
        <ProgressBar
          className="progress--total"
          tooltipText="Total number of episodes"
        />
      </div>
    </div>
  );
});

WatchingProgress.propTypes = {};

export default WatchingProgress;
