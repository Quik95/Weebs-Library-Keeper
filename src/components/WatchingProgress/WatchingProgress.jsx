import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';

// Custom components
import ProgressBar from './ProgressBar';

// styles
import useStyles from './WatchingProgress.styles';

const WatchingProgress = memo(function WatchingProgress(props) {
  const classes = useStyles();

  return (
    <div className={classes.watchingProgress}>
      <Typography variant="caption">10/15</Typography>
      <ProgressBar />
    </div>
  );
});

WatchingProgress.propTypes = {};

export default WatchingProgress;
