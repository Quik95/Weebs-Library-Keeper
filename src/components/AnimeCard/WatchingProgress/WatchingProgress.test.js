import React from 'react';
import renderer from 'react-test-renderer';

import WatchingProgress from './WatchingProgress';

describe('<WatchingProgress/>', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<WatchingProgress />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
