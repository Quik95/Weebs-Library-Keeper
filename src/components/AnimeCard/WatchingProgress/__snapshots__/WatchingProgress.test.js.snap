// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WatchingProgress/> Renders correctly 1`] = `
<div
  className="makeStyles-watchingProgress-6"
  data-testid="WatchingProgress"
>
  <span
    className="MuiTypography-root MuiTypography-caption"
  >
    10/15
  </span>
  <div
    className="makeStyles-progressBarContainer-1"
  >
    <div
      aria-describedby={null}
      className="makeStyles-progress-2 makeStyles-progress--watched-4"
      data-testid="ProgressBar"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Number of episodes you watched"
    />
    <div
      aria-describedby={null}
      className="makeStyles-progress-2 makeStyles-progress--aired-5"
      data-testid="ProgressBar"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Number of episodes aired"
    />
    <div
      aria-describedby={null}
      className="makeStyles-progress-2 makeStyles-progress--total-3"
      data-testid="ProgressBar"
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="Total number of episodes"
    />
  </div>
</div>
`;
